### ISSUE TITLE
False positive for SpaceBeforeBlockParameters: false with new lambda syntax

### ISSUE THREAD

**Issue by padde** [rubocop/rubocop#1197]
``` yaml
# .rubocop.yml

Style/SpaceInsideBlockBraces:
  SpaceBeforeBlockParameters: false
```

checking the following code:

``` ruby
->(x) { x }
```

produces a false positive:

```
Inspecting 1 file
C

Offenses:

test.rb:1:8: C: Space between { and | detected.
->(x) { x }
```
